api/src/main/java/io/kaoto/backend/api/resource/v2/IntegrationsResource.java

```diff
@@ -0,0 +1,152 @@
+import org.eclipse.microprofile.openapi.annotations.parameters.RequestBody;
+import org.jboss.logging.Logger;
+import org.jboss.resteasy.reactive.server.ServerExceptionMapper;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * 🐱class IntegrationsResource
+ * 🐱relationship compositionOf DeploymentService, 0..1
+ *
+ * This endpoint will return the yaml needed to deploy
+ * the related integration and the
+ * endpoints to interact with deployments.
+ */
+@Path("/v2/integrations")
+@ApplicationScoped
+public class IntegrationsResource {
+
+    private final Logger LOG = Logger.getLogger(IntegrationsResource.class);
+
+    @Inject
+    public void setDeploymentService(
+            final DeploymentService deploymentService) {
+        this.deploymentService = deploymentService;
+    }
+    private DeploymentService deploymentService;
+
+    @Inject
+    public void setStepParserServices(final Instance<StepParserService<Step>> stepParserServices) {
+        this.stepParserServices = stepParserServices;
+    }
+    private Instance<StepParserService<Step>> stepParserServices;
+
+    /*
+     * 🐱method CRDs: Map
+     * 🐱param dsl: String
+     * 🐱param integration: List<Integration>
+     *
+     * Idempotent operation that given an array of integrations, returns the corresponding CRDs.
+     */
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces("text/yaml")
+    @Path("/")
+    @CacheResult(cacheName = "api")
+    @Operation(summary = "Get CRDs",
+            description = "Returns the associated custom resource definitions. This is an idempotent operation.")
+    public String crds(
+            final @RequestBody List<Integration> request,
+            final @Parameter(description = "DSL to use. For example: 'Kamelet Binding'.")
+            @QueryParam("dsl") String dsl) {
+        return deploymentService.crds(request, dsl);
+    }
+
+    /*
+     * 🐱method integration: Map
+     * 🐱param dsl: String
+     * 🐱param crd: String
+     *
+     * Idempotent operation that given a CRD, returns the JSON representation.
+     *
+     */
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes("text/yaml")
+    @Path("/")
+    @CacheResult(cacheName = "api")
+    @Operation(summary = "Get Integration Object",
+            description = "Given the associated custom resource definition, returns the JSON object."
+                    + " This is an idempotent operation.")
+    public List<Integration> integration(
+            final @RequestBody String crd,
+            final @Parameter(description = "DSL to use. For example: "
+                    + "'Kamelet Binding'.")
+            @QueryParam("dsl") String dsl) {
+        List<Integration> integrations = new ArrayList<>();
```

Comment:
We should move all this to some service and keep the resources classes clean, but better not add more changes, it is already very long.